### PR TITLE
Tunnel message: add length field

### PR DESCRIPTION
**Pull request by olliw42**
addresses #1220

EDITED (HamishW). The tunnel message is WIP, and Ollie is the person who championed it and is first implementer. This change makes sense and is not affecting stable messages.

EDIT (OlliW): Thx for the edit! I should have realized that it's important info that changing the message now won't do damage.

### CODE REVIEW

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -5727,7 +5727,8 @@
...
-      <field type="uint8_t[128]" name="payload">Variable length payload. The payload length is defined by the remaining message length when subtracting the header and other fields. The entire content of this block is opaque unless you understand the encoding specified by payload_type.</field>
+      <field type="uint8_t" name="payload_len">Length of the data transported in payload</field>
+      <field type="uint8_t[128]" name="payload">Variable length payload. The payload length is defined by payload_len. The entire content of this block is opaque unless you understand the encoding specified by payload_type.</field>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: Is it worth revisiting the length of this array at the same time - e.g. making it max available?

File: message_definitions/v1.0/common.xml [hunk 2]
```diff
@@ -5727,7 +5727,8 @@
...
       <field type="uint8_t" name="target_component">Component ID (can be 0 for broadcast, but this is discouraged)</field>
       <field type="uint16_t" name="payload_type" enum="MAV_TUNNEL_PAYLOAD_TYPE">A code that identifies the content of the payload (0 for unknown, which is the default). If this code is less than 32768, it is a 'registered' payload type and the corresponding code should be added to the MAV_TUNNEL_PAYLOAD_TYPE enum, and the entry possibly to https://github.com/mavlink/mavlink/tunnel-message-payload-types.xml. Software creators can register blocks of types as needed. Codes greater than 32767 are considered local experiments and should not be checked in to any widely distributed codebase.</field>
-      <field type="uint8_t[128]" name="payload">Variable length payload. The payload length is defined by the remaining message length when subtracting the header and other fields. The entire content of this block is opaque unless you understand the encoding specified by payload_type.</field>
+      <field type="uint8_t" name="payload_len">Length of the data transported in payload</field>
```
Comment (hamishwillee) [inline message_definitions/v1.0/common.xml]: I'd slightly prefer `payload_length` as name.

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (olliw42) [reply message_definitions/v1.0/common.xml]: I personally (still) don't see a need for it
and for the same argument as before of keeping resources lower I (still) prefer 128 bytes max

File: message_definitions/v1.0/common.xml (hunk 2 above)
Comment (olliw42) [reply message_definitions/v1.0/common.xml]: yeah ... I checked with the other messages which have a length/size field, and the naming isn't really consistent ... payload_length is fine to me, would you maybe prefer payload_size? I really have no opinion.

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (mhkabir) [reply message_definitions/v1.0/common.xml]: What is the argument behind not making it the max available? In practice, won't the truncation cut out whatever is zero at the end?

File: message_definitions/v1.0/common.xml (hunk 2 above)
Comment (olliw42) [reply message_definitions/v1.0/common.xml]: ok, so changed to payload_length :)

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml]: @mhkabir Truncation affects what is sent OTA, but not the resources allocated to handle the message either end. Ollie is working on gimbals, so smaller is better for him.

**Comment by hamishwillee** [comment]
Thanks @olliw42 . Merged.

**Pull request merged**